kotest-framework/kotest-framework-plugin-gradle/src/main/kotlin/io/kotest/framework/gradle/KotestPlugin.kt

```diff
@@ -30,42 +34,58 @@ open class KotestPlugin : Plugin<Project> {
    override fun apply(project: Project) {
 
       // allows users to configure the test engine
-      val extension = project.extensions.create(
-         /* name = */ EXTENSION_NAME,
-         /* type = */ KotestExtension::class.java,
-         /* ...constructionArguments = */ project
-      )
+      val kotestExtension = project.extensions.create<KotestExtension>(EXTENSION_NAME)
+
+      configureTaskConventions(project)
 
       // Configure Kotlin JVM projects
+      handleKotlinJvm(project)
+
+      // Configure Kotlin multiplatform projects
+      handleKotlinMultiplatform(project)
+
+      // Configure Kotlin Android projects
+      handleKotlinAndroid(project)
+   }
+
+   private fun configureTaskConventions(project: Project) {
+      project.tasks.withType<AbstractKotestTask>().configureEach {
```

Comment:
this works even though you add the tasks after?

Reply:
Yes. The block inside `configureEach {}` will be invoked lazily, only when Gradle creates each task with a matching type.

Reply:
👍🏻